src/qibolab/transpilers/transpile.py

```diff
@@ -66,19 +91,26 @@ def can_execute(circuit, verbose=True):
 
         if len(gate.qubits) == 1:
             if not isinstance(gate, (gates.I, gates.Z, gates.RZ, gates.U3)):
-                if verbose:
-                    vlog(f"{gate.name} is not native gate.")
+                vlog(f"{gate.name} is not a single qubit native gate.")
                 return False
 
         elif len(gate.qubits) == 2:
-            if not isinstance(gate, gates.CZ):
-                vlog(f"{gate.name} is not native gate.")
+            if CZ_is_native and iSWAP_is_native:
+                if not isinstance(gate, (gates.CZ, gates.iSWAP)):  # pragma: no cover
+                    vlog(f"{gate.name} is not a two qubit native gate.")
+                    return False
+            elif CZ_is_native:
+                if not isinstance(gate, (gates.CZ)):  # pragma: no cover
+                    vlog(f"{gate.name} is not a two qubit native gate.")
+                    return False
+            elif iSWAP_is_native:
+                if not isinstance(gate, (gates.iSWAP)):  # pragma: no cover
+                    vlog(f"{gate.name} is not a two qubit native gate.")
+                    return False
+            else:  # pragma: no cover
+                vlog("Use only CZ and/or iSWAP as native gates")
                 return False
```

Comment:
I believe this can be done simpler using just
```suggestion
            if gate.__class__.__name__ not in two_qubit_natives:                    
                vlog(f"{gate.name} is not a two qubit native gate.")
                return False
```